### PR TITLE
change to mongoDb and get animals by filter

### PR DESCRIPTION
**Pull request by KaabiTaoufik**

### CODE REVIEW

File: backend/src/app.module.ts [hunk 1]
```diff
@@ -7,19 +7,28 @@ import { UserModule } from './user/user.module';
 import { AnimalModule } from './animal/animal.module';
 import { BreedingRequestModule } from './breeding-request/breeding-request.module';
 import configuration from './config/configuration';
+import { MongooseModule } from '@nestjs/mongoose';
 
 @Module({
   imports: [
+    TypeOrmModule.forRoot({
```
Comment (alabenhamouda) [inline backend/src/app.module.ts]: why not changing the object returned by the configuration function

File: backend/src/app.module.ts [hunk 2]
```diff
@@ -7,19 +7,28 @@ import { UserModule } from './user/user.module';
...
+    ],
+    ssl: true,
+    useUnifiedTopology: true,
+    useNewUrlParser: true,
+    autoLoadEntities: true,
+    synchronize: true, // remove in prod
+    }),
     ConfigModule.forRoot({
       isGlobal: true,
-      envFilePath: '.env.local',
```
Comment (alabenhamouda) [inline backend/src/app.module.ts]: undo, .env is just a template and real credentials should be placed in the ignored filed .env.local

File: backend/src/app.module.ts [hunk 3]
```diff
@@ -7,19 +7,28 @@ import { UserModule } from './user/user.module';
...
+      envFilePath: '.env',
       load: [configuration],
     }),
-    TypeOrmModule.forRootAsync({
```
Comment (alabenhamouda) [inline backend/src/app.module.ts]: shoud be kept

File: backend/src/animal/entities/animal.entity.ts [hunk 4]
```diff
@@ -8,11 +8,13 @@ import {
   ManyToOne,
   PrimaryGeneratedColumn,
   OneToMany,
+  ObjectIdColumn,
 } from 'typeorm';
 
 @Entity('animals')
 export class Animal extends Timestamp {
   @PrimaryGeneratedColumn('uuid')
```
Comment (alabenhamouda) [inline backend/src/animal/entities/animal.entity.ts]: to remove since we use ObjectIdColumn

**Review by alabenhamouda — CHANGES_REQUESTED** [review]

File: backend/src/app.module.ts (hunk 1 above)
Comment (KaabiTaoufik) [reply backend/src/app.module.ts]: what do you mean exactly ? ( i followed the documentation of nestjs)

Comment (alabenhamouda) [reply backend/src/app.module.ts]: the configuration is outsourced in the config/configuration.ts file which holds the database information read from the .env.local file

**Review by alabenhamouda — APPROVED** [review]

**Pull request merged**